### PR TITLE
Docs footer

### PR DESCRIPTION
**Pull request by cdedreuille**
- Add Feedback form
- Add edit on Github button

### CODE REVIEW

File: apps/frontpage/components/docs/footer/actions.ts [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+      if (node.type === 'directory') {
+        addSlugs(node.children!);
+      } else {
+        slugs.add(node.slug);
+      }
+    });
+  }
+
+  versions.forEach((v) => {
+    const tree = generateDocsTree(`content/docs/${v}`);
```
Comment (kylegach) [inline apps/frontpage/components/docs/footer/actions.ts:63]: This doesn't work, because that location doesn't exist:

```
Error: ENOENT: no such file or directory, scandir 'content/docs/8.0-test-1'
```
— [Vercel log](https://vercel.com/storybookjs/web/logs?slug=app-future&slug=en-US&slug=storybookjs&slug=web&slug=logs&page=1&timeline=past30Minutes&startDate=1717963701744&endDate=1717965501744&selectedLogId=1717965490049549004920600000&selectedLogTimestamp=1717965490049&forceEndDate=1717965490049)

It may only be a matter of using `path` & `__dirname`?

File: apps/frontpage/lib/get-tree.ts [hunk 2]
```diff
@@ -27,13 +27,15 @@ export const generateDocsTree = (pathToFiles?: string, docsRoot?: string) => {
   const newPath = pathToFiles || 'content/docs';
   const newDocsRoot = docsRoot || newPath;
 
-  const files = fs.readdirSync(newPath);
```
Comment (kylegach) [inline apps/frontpage/lib/get-tree.ts:30]: This path could not be read when deployed without this change.

See more: https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-process.cwd()

File: apps/frontpage/components/docs/footer/form.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
```
Comment (kylegach) [inline apps/frontpage/components/docs/footer/form.tsx:1]: Note the lack of a `use "client"` directive.

TL;DR: That directive is only needed at the server/client boundary. Including it here, which is always rendered inside Footer and therefore always client, caused TS errors.

See [this comment](https://github.com/vercel/next.js/discussions/46795#discussioncomment-5248407) and [this reply](https://github.com/vercel/next.js/discussions/46795#discussioncomment-5937396) for details.

File: apps/frontpage/lib/get-tree.ts (hunk 2 above)
Comment (kylegach) [reply apps/frontpage/lib/get-tree.ts:30]: Damn. Still doesn't work. [Logs](https://vercel.com/storybookjs/web/logs?slug=app-future&slug=en-US&slug=storybookjs&slug=web&slug=logs&page=1&timeline=past30Minutes&startDate=1718738690922&endDate=1718740490922&selectedLogId=171874047458047458096300000&selectedLogTimestamp=1718740474580&forceEndDate=1718740476165)

File: apps/frontpage/next.config.js [hunk 4]
```diff
@@ -74,6 +74,11 @@ module.exports = {
     return config;
   },
   transpilePackages: ['@repo/ui', '@repo/utils'],
+  experimental: {
```
Comment (kylegach) [inline apps/frontpage/next.config.js:77]: See: https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-next.js
